packages/nhsuk-frontend/src/nhsuk/components/file-upload/file-upload.mjs

```diff
@@ -0,0 +1,492 @@
+import { closestAttributeValue } from '../../common/closest-attribute-value.mjs'
+import { ConfigurableComponent } from '../../configurable-component.mjs'
+import { ElementError } from '../../errors/index.mjs'
+import { I18n } from '../../i18n.mjs'
+
+/**
+ * File upload component
+ *
+ * @preserve
+ * @augments ConfigurableComponent<FileUploadConfig>
+ */
+export class FileUpload extends ConfigurableComponent {
+  /**
+   * @private
+   * @type {HTMLFileInputElement}
+   */
+  $input
+
+  /**
+   * @private
+   */
+  $button
+
+  /**
+   * @private
+   */
+  $status
+
+  /** @private */
+  i18n
+
+  /** @private */
+  id
+
+  /** @private */
+  $announcements
+
```

Comment:
We use a newer TypeScript version so can remove these

Only need optional class fields that aren't added in the constructor (it can infer the others itself)

```suggestion
```